uint32 flag
uint32 error_code
string error_msg
string boot_msg
uint32 progress_level
string[] types
float32[] confidences
